### PR TITLE
🔧 Add `CREATE3` Tests

### PR DESCRIPTION
**Pull request by pcaversaccio**
### 🕓 Changelog

> This PR is rebased on [#17](https://github.com/pcaversaccio/createx/pull/17).

This PR adds the tests for all `CREATE3` functions in `CreateX`; namely:
- `function computeCreate3Address(bytes32,address) pure returns (address)`
- `function computeCreate3Address(bytes32) view returns (address)`
- `function deployCreate3(bytes) payable returns (address)`
- `function deployCreate3(bytes32,bytes) payable returns (address)`
- `function deployCreate3AndInit(bytes32,bytes,bytes,tuple(uint256,uint256)) payable returns (address)`
- `function deployCreate3AndInit(bytes,bytes,tuple(uint256,uint256)) payable returns (address)`
- `function deployCreate3AndInit(bytes32,bytes,bytes,tuple(uint256,uint256),address) payable returns (address)`
- `function deployCreate3AndInit(bytes,bytes,tuple(uint256,uint256),address) payable returns (address)`

Further changes include:
- Fix a small bug regarding the `salt` parameter handling in `deployCreate3` and `deployCreate3AndInit` in `CreateX`;
- Add a section in the `README` about the 100% test coverage as well as add a coverage badge;
- Add Scroll Sepolia testnet & mainnet, Base Sepolia, and Optimism Sepolia deployment configurations;
- Add CI pipeline for coverage;
- Bump `hardhat`, `@nomicfoundation/hardhat-verify`, `eslint`, `solhint`, `@typescript-eslint/eslint-plugin`, and `@typescript-eslint/parser` to the latest version;
- Bump submodules;

**NOTE:** Since I use the latest `@nomicfoundation/hardhat-verify` version, which supports natively Sourcify verification, we can verify Etherscan, Blockscout (which syncs from Sourcify) _and_ Sourcify in one run:

image

#### PR Checklist

- [ ] Because this PR includes a **bug fix**, relevant tests have been included.
- [x] Because this PR includes a **new feature**, the change was previously discussed in an [issue](https://github.com/pcaversaccio/createx/issues) or in the [discussions](https://github.com/pcaversaccio/createx/discussions) section. (See https://github.com/pcaversaccio/createx/issues/10.)
- [ ] I didn't do anything of this.

#### 🐶 Cute Animal Picture

image

### CODE REVIEW

**Review by pcaversaccio — COMMENTED** [review]
IMPORTANT: @mds1 I enabled automatic squash-merge. So if you approve this PR without leaving any unresolved messages, it will be automatically merged.

File: package.json [hunk 1]
```diff
@@ -22,14 +22,13 @@
   "author": "pcaversaccio (https://pcaversaccio.com), Matt Solomon (https://mattsolomon.dev)",
   "license": "AGPL-3.0-only",
   "scripts": {
-    "help": "npx hardhat help",
-    "accounts": "npx hardhat accounts",
-    "balances": "npx hardhat balances",
-    "node": "npx hardhat node",
-    "clean": "npx hardhat clean",
+    "clean:hh": "npx hardhat clean",
+    "clean:forge": "forge clean",
```
Comment (mds1) [inline package.json:26]: What are the use cases for two separate commands, as opposed to just
```suggestion
    "clean": "npx hardhat clean && forge clean",
```

File: package.json [hunk 2]
```diff
@@ -22,14 +22,13 @@
...
     "test": "forge test --out forge-artifacts",
-    "coverage": "npx hardhat coverage",
-    "compile": "npx hardhat compile",
+    "snapshot": "forge snapshot --out forge-artifacts",
+    "coverage": "forge coverage --report summary",
+    "coverage:report": "forge coverage --report lcov && lcov --remove lcov.info 'test/*' --output-file lcov.info --rc lcov_branch_coverage=1 && genhtml lcov.info --branch-coverage --output-dir coverage",
+    "compile:hh": "npx hardhat compile",
     "build": "forge build --out forge-artifacts",
```
Comment (mds1) [inline package.json:32]: Similar to above, plus would be nice for these to both be called `compile` or both `build`

File: package.json (hunk 1 above)
Comment (pcaversaccio) [reply package.json]: implemented  via https://github.com/pcaversaccio/createx/pull/18/commits/b8c3646a295418d978ea321ceee83aeb72a907fb

File: package.json (hunk 2 above)
Comment (pcaversaccio) [reply package.json]: Implemented via https://github.com/pcaversaccio/createx/pull/18/commits/b8c3646a295418d978ea321ceee83aeb72a907fb.

**Review by mds1 — APPROVED** [review]
Looks great, really thorough job!

**Pull request merged**